trade-finance-manager-ui/component-tests/utilisation-reports/utilisation-report-reconciliation-for-bank.component-test.js

```diff
@@ -0,0 +1,43 @@
+const { PRIMARY_NAVIGATION_KEYS } = require('../../server/constants');
+const pageRenderer = require('../pageRenderer');
+
+const page = '../templates/utilisation-reports/utilisation-report-reconciliation-for-bank.njk';
+const render = pageRenderer(page);
+
+describe(page, () => {
+  let wrapper;
+
+  const bank = {
+    id: '123',
+    name: 'Test Bank',
+  };
+  const formattedReportPeriod = 'Nov 2023';
+
+  const params = {
+    activePrimaryNavigation: PRIMARY_NAVIGATION_KEYS.UTILISATION_REPORTS,
+    bank,
+    formattedReportPeriod,
+  };
+
+  beforeEach(() => {
+    wrapper = render(params);
+  });
+
+  it('should render the main heading', () => {
+    wrapper.expectElement('[data-cy="utilisation-report-reconciliation-for-bank-heading"]').toExist();
+  });
+
+  it('should render the report period heading', () => {
+    wrapper.expectText('[data-cy="report-period-heading"]').toRead(`${formattedReportPeriod}`);
+  });
+
+  it.each`
+    tabName               | dataCy
+    ${'Premium payments'} | ${'bank-report-tab-premium-payments'}
+    ${'Keying sheet'}     | ${'bank-report-tab-keying-sheet'}
+    ${'Payment details'}  | ${'bank-report-tab-payment-details'}
+    ${'Utilisation'}      | ${'bank-report-tab-utilisation'}
+  `("should render the '$tabName' tab", ({ dataCy }) => {
+    wrapper.expectElement(`[data-cy="${dataCy}"]`).toExist();
```

Comment:
I can see we use lots of `data-cy` on the project for our component tests. There is an issue with this approach: we are just testing that we have put certain test-helper attributes in place and not actually testing anything of real substance.
In this case, we obviously don't want to be testing the gov uk component so I think find to still use the same selector, but maybe we could check the text content of the element returned contains the tab name so that we are testing we have set the tab names correctly?

Reply:
I think this was one of the things I added last December hence it lacking any real substance. I updated this test in FN-1615 with the same ideas you've suggested, but I'll move those updates to this PR :)